src/api/cvc4cpp.h

```diff
@@ -1818,129 +1857,87 @@ class CVC4_PUBLIC Solver
    */
   Term mkBoolean(bool val) const;
 
-  /**
-   * Create an Integer constant.
-   * @param s the string represetntation of the constant
-   * @param base the base of the string representation
-   * @return the Integer constant
-   */
-  Term mkInteger(const char* s, uint32_t base = 10) const;
-
-  /**
-   * Create an Integer constant.
-   * @param s the string represetntation of the constant
-   * @param base the base of the string representation
-   * @return the Integer constant
-   */
-  Term mkInteger(const std::string& s, uint32_t base = 10) const;
-
-  /**
-   * Create an Integer constant.
-   * @param val the value of the constant
-   * @return the Integer constant
-   */
-  Term mkInteger(int32_t val) const;
-
-  /**
-   * Create an Integer constant.
-   * @param val the value of the constant
-   * @return the Integer constant
-   */
-  Term mkInteger(uint32_t val) const;
-
-  /**
-   * Create an Integer constant.
-   * @param val the value of the constant
-   * @return the Integer constant
-   */
-  Term mkInteger(int64_t val) const;
-
-  /**
-   * Create an Integer constant.
-   * @param val the value of the constant
-   * @return the Integer constant
-   */
-  Term mkInteger(uint64_t val) const;
-
   /**
    * Create a constant representing the number Pi.
    * @return a constant representing Pi
    */
   Term mkPi() const;
 
   /**
-   * Create an Real constant.
-   * @param s the string represetntation of the constant
-   * @param base the base of the string representation
-   * @return the Real constant
+   * Create a real constant.
+   * @param s the string representation of the constant
+   * @return a constant of sort Real or Integer (if the string is an integer)
    */
-  Term mkReal(const char* s, uint32_t base = 10) const;
+  Term mkReal(const char* s) const;
```

Comment:
Why did you remove the `base` argument?

Reply:
The reason is that we don't support constructing real numbers from something like `101.010` (binary base). The `Rational(const std::string& s, unsigned base)` constructor expects a string of the form `1101/101` (binary base). For the parser, I needed to support parsing constants like `9.123`, so I changed `mkReal()` to use `Rational::fromDecimal()` and removed the `base` parameter. We could split this up into two constructors `mkReal()` and `mkRational()` though but I wasn't sure if we really need that.

Reply:
For the record: we have decided to support only the decimal base but both styles `123/456` and `123.456`.